src/main/scala/tech/cryptonomic/conseil/tezos/TezosNodeInterface.scala

```diff
@@ -100,4 +131,49 @@ object TezosNodeInterface extends TezosRPCInterface with LazyLogging {
     }
   }
 
+  /** connection pool settings customized for streaming requests */
+  private[this] val streamingRequestsConnectionPooling = ConnectionPoolSettings(
+    conf
+      .atPath("akka.tezos-streaming-client.connection-pool")
+      .withFallback(ConfigFactory.defaultReference())
+  )
+
+  /** creates a connections pool based on the host network */
+  private[this] def hostPool(network: String) = {
+    val host = conf.getString(s"platforms.tezos.$network.node.hostname")
+    val port = conf.getInt(s"platforms.tezos.$network.node.port")
+    val protocol = conf.getString(s"platforms.tezos.$network.node.protocol")
+    if (protocol == "https")
+      Http().cachedHostConnectionPoolHttps[String](
+        host = host,
+        port = port,
+        settings = streamingRequestsConnectionPooling
+      )
+    else
+      Http().cachedHostConnectionPool[String](
+        host = host,
+        port = port,
+        settings = streamingRequestsConnectionPooling
+      )
+  }
+
+
+
+  override def runBatchedGetQuery(network: String, commands: List[String]): Future[List[String]] = {
```

Comment:
please refer to my reply on comment at `TezosNodeOperator.getAllAccountsForBlock`, down below

Reply:
As a thumb rule I'd use the batched call when a potentially significant number of http requests to tezos need to be made concurrently (about 20+), so that each requests will be queued for the bare minimum required, as defined in the `tezos-streaming-client` host pool in `application.conf`